ipa-core/src/protocol/ipa_prf/mod.rs

```diff
@@ -171,10 +172,13 @@ where
     C: UpgradableContext,
     C::UpgradedContext<Boolean>: UpgradedContext<Boolean, Share = Replicated<Boolean>>,
     C::UpgradedContext<F>: UpgradedContext<F, Share = Replicated<F>>,
-    BK: SharedValue + U128Conversions + CustomArray<Element = Boolean> + Field,
-    TV: SharedValue + U128Conversions + CustomArray<Element = Boolean> + Field,
-    TS: SharedValue + U128Conversions + CustomArray<Element = Boolean> + Field,
-    SS: SharedValue + U128Conversions + CustomArray<Element = Boolean> + Field,
+    BK: SharedValue + U128Conversions + CustomArray<Element = Boolean>,
+    TV: SharedValue + U128Conversions + CustomArray<Element = Boolean>,
+    TS: SharedValue + U128Conversions + CustomArray<Element = Boolean>,
+    SS: SharedValue + U128Conversions + CustomArray<Element = Boolean>,
+    Replicated<BK>: BooleanArrayMul,
```

Comment:
I guess you tried it already, but I am going to ask anyway - have you tried to provide blanket implementation `BooleanArrayMul` for all boolean shares?

Reply:
It's another consequence of const generic limitations -- see [this comment](https://github.com/private-attribution/ipa/pull/958/files#diff-17a4607905850bd30e0208dce5fc0540e1ec37fbfce53cdb60aa5858de93d807R56-R60).

Part of why things get weird here is that we're using the record-wise vectorization support for bit-wise vectorization. To make a fully vectorized version of these protocols we'd need to change that, and could probably get rid of BooleanArrayMul in the process.